lib/certlint/extensions/keyusage.rb

```diff
@@ -47,7 +47,7 @@ def self.lint(content, cert, critical = false)
           'CRL Sign'
         ]
         if v.any? { |u| !allowed.include? u }
-          messages << 'E: Unallowed key usage for RSA public key'
+          messages << "E: Unallowed key usage for RSA public key (#{(u-v).join(', ')})"
```

Comment:
`u` isn't a valid variable at this point.  I think you want `v - allowed` instead of `u-v`.  This test could probably be changed to something like:
```ruby
unallowed = v - allowed
if !unallowed.empty?
```

The same is true for the other two changes.